### PR TITLE
Hide Pagination When Only One Page of Content Exists; closes #2566.

### LINKED ISSUE #2566: Hide Pagination When Only One Page of Content Exists
title says it all, if you have less entities in a list than the default per page setting, then the pagination controls should be hidden. If the per-page value does NOT equal the default, then it should remain visible.

### PR DESCRIPTION
**Pull request by nicolad**
## Problem this Pull Request solves
- Closes: https://github.com/eventespresso/event-espresso-core/issues/2566
- As saying goes: title says it all.

## How has this been tested

Peek 2020-03-20 14-20

### CODE REVIEW

File: assets/src/application/ui/layout/entityList/EntityList.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ const EntityList = <E extends Entity, ELFS extends EntityListFilterStateManager<
 	const error = isError(entityType);
 	const loading = isLoading(entityType);
 
-	const filteredEntities = useFilteredEntities(domain, listId, entities, filterState);
+	const { paginatedEntities, searchResults } = useFilteredEntities(domain, listId, entities, filterState);
```
Comment (nicolad) [inline assets/src/application/ui/layout/entityList/EntityList.tsx]: I am not :100: happy that `useFilteredEntities` is now leaking it's internals, open to suggestions

File: assets/src/application/ui/layout/entityList/pagination/EntityPagination.tsx [hunk 2]
```diff
@@ -14,25 +17,29 @@ interface EntityPaginationProps<ELFS extends EntityListFilterStateManager> exten
...
 	pageSizeOptions = ['2', '6', '12', '24', '48'],
 	showSizeChanger = true,
 	showTotal,
-	filterState,
 	...rest
 }) => {
-	const { total, setPageNumber, setPerPage, pageNumber }: EntityListFilterStateManager = filterState;
+	const { pageNumber, perPage, setPerPage, setPageNumber, total }: EntityListFilterStateManager = filterState;
+
+	const className = classNames('ee-entity-pagination', { hidePagination: perPage >= entities.length });
```
Comment (manzoorwanijk) [inline assets/src/application/ui/layout/entityList/pagination/EntityPagination.tsx]: You can simply do this:
```suggestion
	const className = classNames('ee-entity-pagination', { hidePagination: total <= perPage });
```

Comment (nicolad) [reply assets/src/application/ui/layout/entityList/pagination/EntityPagination.tsx]: updated, thx

**Comment by nicolad** [comment]
ezgif com-crop

File: assets/src/application/ui/layout/entityList/pagination/EntityPagination.tsx [hunk 3]
```diff
@@ -14,25 +15,30 @@ interface EntityPaginationProps<ELFS extends EntityListFilterStateManager> exten
...
 	pageSizeOptions = ['2', '6', '12', '24', '48'],
 	showSizeChanger = true,
 	showTotal,
-	filterState,
 	...rest
 }) => {
-	const { total, setPageNumber, setPerPage, pageNumber }: EntityListFilterStateManager = filterState;
+	const { pageNumber, perPage, setPerPage, setPageNumber, total }: EntityListFilterStateManager = filterState;
+
+	console.log({ perPage, total });
```
Comment (tn3rb) [inline assets/src/application/ui/layout/entityList/pagination/EntityPagination.tsx]: booooooo 😆

Comment (nicolad) [reply assets/src/application/ui/layout/entityList/pagination/EntityPagination.tsx]: it's embarrassante, fixed

**Pull request merged**